art/defences/detector/poison/ground_truth_evaluator.py

```diff
@@ -40,7 +40,7 @@ def __init__(self):
         """
 
     def analyze_correctness(
-        self, assigned_clean_by_class: np.ndarray, is_clean_by_class: list
+        self, assigned_clean_by_class: Union[np.ndarray, List[np.ndarray]], is_clean_by_class: list
```

Comment:
Why is the list of np.ndarray type required?

Reply:
The `Union` operator allows either of the types inside. This change allows both np arrays and lists of np arrays to be accepted. This is helpful because the output of another util method, `segment_by_class`, returns a list of np arrays.

Because of the way the function enumerates through the `assigned_clean_by_class` parameter, the output is actually the same for lists of np arrays and np arrays formatted correctly from lists. This allows us to remove an extra step of processing by using the lists as is.